druid-shell/src/backend/wayland/surfaces/surface.rs

```diff
@@ -469,6 +478,26 @@ impl Data {
         self.schedule_deferred_task(DeferredTask::Paint);
     }
 
+    fn open_file(&self, options: FileDialogOptions) -> Option<FileDialogToken> {
+        // FIXME: current ashpd has issues with wayland versions
```

Comment:
is there a link to the upstream issue we could put here as a reference?

Reply:
I guess [this](https://github.com/bilelmoussaoui/ashpd/pull/76). Which is merged, but there's still an issue with `asphd`'d wayland-client version not matching ours (and [depending on a pre-release](https://github.com/bilelmoussaoui/ashpd/pull/79), which is a bit of a pain). I think the sanest thing to do is to wait until version 0.3 of the wayland-client libraries.

Also, it looks like getting an identifier requires a round-trip to the compositor, and so the new function is async...